cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -1225,6 +1227,99 @@ public void getInstances(FullHttpRequest request, HttpResponder responder,
+   *  {"appId": "App2", "programType": "Workflow", "programId": "DataPipelineWorkflow"}]
+   * </code></pre>
+   * </p><p>
+   * </p><p>
+   * The response will be an array of JsonObjects each of which will contain the three input parameters
+   * as well as 2 fields, "count" which maps to the count of the program and "statusCode" which maps to the
+   * status code for the data in that JsonObjects.
+   * </p><p>
+   * If an error occurs in the input (for the example above, workflow in app1 does not exist),
+   * then all JsonObjects for which the parameters have a valid status will have the count field but all JsonObjects
+   * for which the parameters do not have a valid status will have an error message and statusCode.
+   * </p><p>
+   * For example, if there is no workflow in App1 in the data above, then the response would be 200 OK with following
+   * possible data:
+   * </p>
+   * <pre><code>
+   * [{"appId": "App1", "programType": "Service", "programId": "Service1",
+   * "statusCode": 200, "runCount": 20},
+   * {"appId": "App1", "programType": "Workflow", "programId": "testWorkflow", "statusCode": 404,
+   * "error": "Program 'testWorkflow' is not found"},
+   *  {"appId": "App2", "programType": "Workflow", "programId": "DataPipelineWorkflow", "runnableId": "Flowlet1",
+   *  "statusCode": 200, "runCount": 300}]
+   * </code></pre>
+   */
+  @POST
+  @Path("/runcount")
+  public void getRunCounts(FullHttpRequest request, HttpResponder responder,
+                           @PathParam("namespace-id") String namespaceId) throws Exception {
+    List<BatchProgram> programs = validateAndGetBatchInput(request, BATCH_PROGRAMS_TYPE);
+    if (programs.size() > 100) {
+      throw new BadRequestException(String.format("%s programs found in the request, the maximum number " +
+                                                    "supported is 100", programs.size()));
+    }
+
+    List<ProgramId> programIds =
+      programs.stream().map(batchProgram -> new ProgramId(namespaceId, batchProgram.getAppId(),
+                                                          batchProgram.getProgramType(),
+                                                          batchProgram.getProgramId())).collect(Collectors.toList());
+    List<BatchProgramCount> counts = new ArrayList<>(programs.size());
+    for (RunCountResult runCountResult : lifecycleService.getProgramRunCounts(programIds)) {
+      ProgramId programId = runCountResult.getProgramId();
+      Exception exception = runCountResult.getException();
+      if (exception == null) {
+        counts.add(new BatchProgramCount(programId, 200, null, runCountResult.getCount()));
+      } else if (exception instanceof NotFoundException) {
+        counts.add(new BatchProgramCount(programId, 404, exception.getMessage(), null));
+      } else if (exception instanceof UnauthorizedException) {
+        counts.add(new BatchProgramCount(programId, 403, exception.getMessage(), null));
+      }
```

Comment:
What if the exception is of some other type?

Reply:
The other batch calls does not care about other type, I mark them as 500